ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -0,0 +1,223 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.data.*;
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.result.*;
+import com.microsoft.azure.kusto.ingest.source.*;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.lang.invoke.MethodHandles;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.sql.ResultSet;
+import java.util.*;
+import java.util.zip.GZIPOutputStream;
+
+public class StreamingIngestClient implements IngestClient {
+
+    private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private AzureStorageClient azureStorageClient;
+    private StreamingIngestProvider streamingIngestProvider;
+    private final List<String> mappingRequiredFormats = Arrays.asList("json", "singlejson", "avro");
+    private static final int STREAM_COMPRESS_BUFFER_SIZE = 1024;
+
+    StreamingIngestClient(ConnectionStringBuilder csb) throws URISyntaxException {
+        log.info("Creating a new StreamingIngestClient");
+        this.streamingIngestProvider = ClientFactory.createStreamingIngestProvider(csb);
+        this.azureStorageClient = new AzureStorageClient();
+    }
+
+    StreamingIngestClient(StreamingIngestProvider streamingIngestProvider) {
+        log.info("Creating a new StreamingIngestClient");
+        this.streamingIngestProvider = streamingIngestProvider;
+        this.azureStorageClient = new AzureStorageClient();
+    }
+
+    @Override
+    public IngestionResult ingestFromFile(FileSourceInfo fileSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
```

Comment:
Ingest from file might be slow but it is a common scenario that we want to support. 
We agree that ingesting from blob is not recommended with streaming ingest but we want to give the same API of ingestion across clients. 
I'll comment it isn't recommended.

Reply:
When saying clients, are you talking about batch vs. streaming?
(a suggestions: log a warning for the non-recommended options so that the user will know)

Reply:
We want to have both the same Interface for different client classes within each SDK (so yes in this case batched vs streaming) and the same functionality across all SDKs.